### PR TITLE
Updates to testing page

### PR DESCRIPTION
**Pull request by ximenesuk**
This PR supports a couple of currently open PRs: https://github.com/openmicroscopy/openmicroscopy/pull/3310 and https://github.com/openmicroscopy/openmicroscopy/pull/3319  It should only be considered for merging once both of those two are merged.

--no-rebase

### CODE REVIEW

**Comment by mtbc** [comment]
Staged at http://www.openmicroscopy.org/site/support/omero5.1-staging/developers/testing.html.

File: omero/developers/testing.txt [hunk 1]
```diff
@@ -362,18 +362,27 @@ To view all available markers the :option:`--markers` option can be used with
 :program:`setup.py test` or :program:`py.test` as detailed in
 :ref:`running-python-tests-directly`.
 
-There are currently two custom markers defined:
+There are a small number of custom markers defined:
```
Comment (mtbc) [inline omero/developers/testing.txt]: I think your "are" does remain correct here, but it's exciting to read the grammar arguments.

File: omero/developers/testing.txt [hunk 2]
```diff
@@ -362,18 +362,27 @@ To view all available markers the :option:`--markers` option can be used with
...
+
+    `broken`
+        Used to mark broken tests. These are tests that fail consistently with no
+        obvious quick fix. Broken tests are excluded from the main integration builds
+        and instead are run in a separate daily build. `broken` markers should have a
+        reason, an associated ticket number or both. If there are multiple associated
+        tickets then a comma-separated list should be used.
+
+    `intermittent`
+        Used to merk tests that fail intermittently. `intermittent` markers should
```
Comment (mtbc) [inline omero/developers/testing.txt]: s/merk/mark/

File: omero/developers/testing.txt [hunk 3]
```diff
@@ -362,18 +362,27 @@ To view all available markers the :option:`--markers` option can be used with
...
+    `long_running`
+        Used to mark tests as long running. These are tests that typically take
+        10 minutes or more and are normally excluded from the daily integration
+        builds.
+
+    `broken`
+        Used to mark broken tests. These are tests that fail consistently with no
+        obvious quick fix. Broken tests are excluded from the main integration builds
+        and instead are run in a separate daily build. `broken` markers should have a
+        reason, an associated ticket number or both. If there are multiple associated
```
Comment (mtbc) [inline omero/developers/testing.txt]: perhaps mention "Trac" explicitly re. tickets

File: omero/developers/testing.txt [hunk 4]
```diff
@@ -362,18 +362,27 @@ To view all available markers the :option:`--markers` option can be used with
...
+        tickets then a comma-separated list should be used.
+
+    `intermittent`
+        Used to merk tests that fail intermittently. `intermittent` markers should
+        have a reason, an associated ticket number or both. If there are multiple
+        associated tickets then a comma-separated list should be used. Tests marked
+        as intermittent are normally included in the daily integration builds.
```
Comment (mtbc) [inline omero/developers/testing.txt]: "normally"? it's a job config option or something?

**Comment by mtbc** [comment]
There remain some `/setup.py test -s ...`, do those need adjusting?

File: omero/developers/testing.txt [hunk 5]
```diff
@@ -459,12 +472,6 @@ plugin :pypi:`pytest-pythonpath` can be used to add paths to
 
 .. program:: py.test
 
-.. option:: -s
-
-    This option allows the standard output to be shown on the console::
-
-        py.test test/integration/test_admin.py -s
```
Comment (mtbc) [inline omero/developers/testing.txt]: no longer offered at all then?

Comment (ximenesuk) [inline omero/developers/testing.txt]: It is still offered by `py.test` but it was here as an example of something your couldn't do using `setup.py` The related PR now lets you use this option that way, which is easier, and so I thought the example no longer relevant and possibly confusing. We might, possibly, want to add another example(s) here.

Comment (ximenesuk) [inline omero/developers/testing.txt]: Ah, I see what you mean, I've not added it above!

File: omero/developers/testing.txt (hunk 1 above)
Comment (ximenesuk) [inline omero/developers/testing.txt]: Hmm, not sure. I had originally changed it to 'three' and then thought better of it. I then tried 'some' but that seemed a bit vague.

File: omero/developers/testing.txt (hunk 2 above)
Comment (ximenesuk) [inline omero/developers/testing.txt]: Hmm, 'murk' might have been more apposite!

File: omero/developers/testing.txt (hunk 3 above)
Comment (ximenesuk) [inline omero/developers/testing.txt]: Yep, makes sense.

File: omero/developers/testing.txt (hunk 4 above)
Comment (ximenesuk) [inline omero/developers/testing.txt]: Yes, I wondered about the wording here. Job config has the other two excluded. I wondered about dropping the 'normally' and just updating the docs if we ever separate out `intermittent` tests. My intention is to wrap these intermittent tests with the flaky decorator so they should always be in the main test build. I'll reword.

**Comment by ximenesuk** [comment]
@mtbc thanks for all the comments. I'll push a revision for tomorrow.

File: omero/developers/testing.txt (hunk 1 above)
Comment (ximenesuk) [inline omero/developers/testing.txt]: I think the main object of the verb is "markers" so "are" is correct, but I'm not a grammarian - maybe @hflynn can advise?

**Comment by mtbc** [comment]
Looks good now, thank you for the adjustments.

**Pull request merged**